### DISCUSSION TITLE
[INFO]how to get absolute path of an uploaded image

### DISCUSSION THREAD

**Discussion by justoverclockl** [directus/directus#12423] [Archive]
Hi, i need info for how to get the absolute url of an image uploaded, because now i get:

`"dish_image": "f19e64a4-5b65-46a6-9a17-b042bf92c6e7",`

### To Reproduce

create a collection, upload an image, result is an hash

<details>
<summary>More Info</summary>

### What version of Directus are you using?

latest

### What version of Node.js are you using?

latest

### What database are you using?

directus cloud version

### What browser are you using?

chrome

### What operating system are you using?

win 11

### How are you deploying Directus?

directus cloud

</details>

**Comment by azrikahar** [comment]
Moved this to discussion as it's more of a question instead of a bug 👍 

Currently the API doesn't return the full URL, but all files does follow the structure of `<your cloud domain>/assets/<image id>` as described in the docs here: https://docs.directus.io/reference/files/#accessing-an-file

In case you do want/suggest this, there is an existing discussion #5928 about this so you can upvote it as well.

**Comment by justoverclockl** [comment]
but at moment it shows 

```
{
"errors": [
{
"message": "You don't have permission to access this.",
"extensions": {
"code": "FORBIDDEN"
}
}
]
}
```

what is the permission to make it public?

**Reply by azrikahar** [reply]
You can set it via Roles & Permissions for the Public role, and granting read permission to `directus_files`. Alternatively, especially to avoid exposing files that you _do not_ want to be accessible publicly, you can use Filter rules to only make it apply to a particular folder(s). Here's a past response on these with short clips: https://github.com/directus/directus/discussions/10920#discussioncomment-1931588. Hope that helps 🙂
